homeworkW12D03/src/ba/bitcamp/homeworkW12D03/task02/PicChangerGUI.java

```diff
@@ -0,0 +1,178 @@
+package ba.bitcamp.homeworkW12D03.task02;
+
+import java.awt.BorderLayout;
+import java.awt.Font;
+import java.awt.Graphics;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.concurrent.LinkedBlockingQueue;
+
+import javax.imageio.ImageIO;
+import javax.swing.JButton;
+import javax.swing.JPanel;
+import javax.swing.UIManager;
+import javax.swing.UnsupportedLookAndFeelException;
+
+/**
+ * 
+ * @author Gordan
+ * 
+ *         Represents main panel that contains image in positive form and
+ *         button. By clicking on the button image is changed in negative form.
+ *         And by clicking again image is returned to the positive form.
+ */
+public class PicChangerGUI extends JPanel implements ActionListener {
+
+	private static final long serialVersionUID = -1592269176478889092L;
+
+	// Declaring main elements.
+	private JPanel panel = new MyPanel();
+	private JButton button = new JButton("Negative image");
+
+	// Declaring list of tasks and list of consumers.
+	private LinkedBlockingQueue<Producer> tasks = new LinkedBlockingQueue<>();
+	private ArrayList<Consumer> consumers = new ArrayList<>();
+
+	// Declaring variable that will read image.
+	private BufferedImage image;
+
+	public PicChangerGUI() {
+
+		// Setting the look of panel according to system look.
+		try {
+			UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
+		} catch (ClassNotFoundException | InstantiationException | IllegalAccessException | UnsupportedLookAndFeelException e2) {
+			e2.printStackTrace();
+		}
+		// Reading image.
+		try {
+			image = ImageIO.read(new File("picture.jpg"));
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+
+		// Adding action listener to the button and setting the font of the
+		// button text.
+		button.addActionListener(this);
+		button.setFont(new Font("Serif", Font.BOLD, 20));
+
+		// Setting the layout of the main panel and adding elements.
+		setLayout(new BorderLayout());
+		add(panel);
+		add(button, BorderLayout.SOUTH);
+	}
+
+	/**
+	 * Divides the tasks into parts and starts consumers.
+	 */
+	public void actionPerformed(ActionEvent e) {
+		// Allocating one row of pixels to one producer and adding producer to
+		// the list.
+		for (int i = 0; i < image.getHeight(); i++) {
```

Comment:
Calling a job definition a `Producer` will not turn this approach into producer/consumer solution.

Here is an example approach how this could be achieved:

#### Producer class

This class will load the image and spawn new tasks into the task queue. Each task represents a row of pixels to handle
Class instance is used in a separate thread

#### Consumer class

Consumer class instance is created for each logical cpu core.
Instances are executed in parallel

Both producer and all of the consumers must run in parallel in order for to qualify for producer/consumer approach. If there is no parallelism, than this is just plain `divide and conquer` approach.